src/python/pants/backend/python/util_rules/pex_test.py

```diff
@@ -607,3 +610,21 @@ def test_additional_inputs(rule_runner: RuleRunner) -> None:
         with zipfp.open("__main__.py", "r") as main:
             main_content = main.read().decode()
     assert main_content[: len(preamble)] == preamble
+
+
+def test_venv_pex_distributions_info(rule_runner: RuleRunner) -> None:
+    venv_pex = create_pex_and_get_all_data(
+        rule_runner, pex_type=VenvPex, requirements=PexRequirements(["requests==2.23.0"])
+    )["pex"]
+    distributions = rule_runner.request(PexDistributionsInfo, [venv_pex])
+    assert len(distributions) == 5
+    assert distributions[0].project_name == "certifi"
+    assert distributions[0].version == "2020.12.5"
+    assert distributions[1].project_name == "chardet"
```

Comment:
I guess this happens to work and will continue to happen to work due to a combination of Pants flooring out at Python  3.7 + json module behavior ... but it seems to make sense to compare an expected `{project_name: version}` dict. Pex certainly makes no guaranty here on order of the json output.

Reply:
The rule sorts this all for a stable cache key.